cabal-install/Distribution/Solver/Modular/Explore.hs

```diff
@@ -60,7 +60,7 @@ backjump (EnableBackjumping enableBj) var initial xs =
           | otherwise                            = f (csAcc `CS.union` cs) cm'
 
     logBackjump :: ConflictSet QPN -> ConflictMap -> ConflictSetLog a
-    logBackjump cs cm = failWith (Failure cs Backjump) (cs, updateCM initial cm)
+    logBackjump cs cm = cm `seq` failWith (Failure cs Backjump) (cs, updateCM initial cm)
```

Comment:
> I'd consider a bang pattern instead of seq for readability.

fixed

> Also, in exploreLog below the cm is only updated when conflict counting is enabled, while it is always updated here. As you suggested in #3960, I think the if countConflicts… would make sense here as well. (Or, if you think that seqing cm is already sufficient, remove the if countConflicts in exploreLog as well, which would IMHO improve readability).

I just made a minimal change to avoid conflicts with your PR, but you're right: it's simpler to always update the map.